ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -102,6 +102,49 @@ internal sealed class ArchiWebHandler : IDisposable {
 			return await UrlPostWithSession(SteamCommunityURL, request, data, referer).ConfigureAwait(false);
 		}
 
+		internal async Task AcceptDigitalGiftCards() {
+			const string requestGifts = SteamStoreURL + "/gifts";
+			HtmlDocument response = (await WebLimitRequest(SteamStoreURL, async () => await WebBrowser.UrlGetToHtmlDocument(requestGifts).ConfigureAwait(false)).ConfigureAwait(false))?.Content;
+			if (response == null) {
+				Bot.ArchiLogger.LogNullError(nameof(response));
+				return;
+			}
+
+			const string requestRedeemGift = "/gifts/0/resolvegiftcard";
+			HtmlNodeCollection nodes = response.DocumentNode.SelectNodes("//div[starts-with(@id,'pending_gift_')]");
+			foreach (string strGID in nodes.Where(node => node?.ChildNodes?.Any(giftNode => giftNode?.Attributes["class"]?.Value?.StartsWith("pending_giftcard") == true) == true).Select(node => node.Attributes["id"]?.Value?.Substring(13))) {
+				if (string.IsNullOrEmpty(strGID)) {
+					Bot.ArchiLogger.LogNullError(nameof(strGID));
+					continue;
+				}
+
+				if (!ulong.TryParse(strGID, out ulong gid) || (gid == 0)) {
+					Bot.ArchiLogger.LogGenericWarning(string.Format(Strings.ErrorParsingObject, nameof(strGID)));
+					continue;
+				}
+
+				if (Bot.HandledGifts.Contains(gid)) {
```

Comment:
This function does far too much and accesses internal bot structures. Instead I'd like to see only requests in AWH (returning either `HtmlDocument` or already parsed `HashSet<ulong>` with gift IDs) and the rest of the logic with checking and handling gifts in main `Bot.cs`. AWH should not be used for excessive logic, `AcceptDigitalGiftCards()` is Bot feature, not web handler feature.